pkg/imageoverrides/templateoverrides.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package imageoverrides
+
+import (
+	"os"
+	"strings"
+)
+
+const (
+	// TemplateOverridePrefix ...
+	TemplateOverridePrefix = "TEMPLATE_OVERRIDE_"
+)
+
+// GetTemplateOverrides reads and formats full image reference from environment variables.
+func GetTemplateOverrides() map[string]string {
+	templateOverrides := make(map[string]string)
+
+	// Iterate through environment variables
+	for _, e := range os.Environ() {
+		key, value := parseEnvVar(e)
+		if key != "" && value != "" {
+			templateOverrides[key] = value
+		}
+	}
+
+	// Check if any overrides were found
+	if len(templateOverrides) > 0 {
+		logf.Info("Found image overrides from environment variables set by operand image prefix")
+	}
+
+	return templateOverrides
+}
+
+// parseEnvVar parses the environment variable and extracts key and value
+func parseEnvVar(envVar string) (key string, value string) {
+	pair := strings.SplitN(envVar, "=", 2)
+	if len(pair) == 2 && strings.HasPrefix(pair[0], TemplateOverridePrefix) {
+		key = strings.ToLower(strings.TrimPrefix(pair[0], TemplateOverridePrefix))
```

Comment:
I just realized I forgot to mention that we should fold the key-name extracted from the env var name to lower case (for consistency with imageOverrides and other maybe subtle reasons) and came here to make a comment and find its already being done.  Great!